library/src/main/java/com/google/android/material/motion/runtime/TargetScope.java

```diff
@@ -85,6 +86,21 @@ void commitPlan(PlanInfo plan) {
     }
   }
 
+  void commitAddNamedPlan(PlanInfo planInfo) {
+    // remove first
+    Performer performer = getNamedPerformer(planInfo);
+    removeNamedPlan(planInfo, performer);
+    // then add
+    if (performer instanceof Performer.NamedPlanPerformance) {
```

Comment:
I think this would be taken care of by the fact that a `ClassCastException` would be thrown from https://github.com/seanoshea/material-motion-runtime-android/blob/issues/34-named-plans/library/src/main/java/com/google/android/material/motion/runtime/TargetScope.java#L147 if the performer for this named plan did not conform to `Performer.NamedPlanPerformance` ?

Would you suggest a different type of exception here? I guess we could throw an `IllegalArgumentException` here with a more descriptive error message if you think that'd be useful?